hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -122,6 +123,15 @@ public void setNeedsBootstrapMerge(boolean needsBootstrapMerge) {
     this.needsBootstrapMerge = needsBootstrapMerge;
   }
 
+  // Getter and Setter for useRecordPosition
+  public boolean getUseRecordPosition() {
+    return useRecordPosition;
+  }
```

Comment:
Rename the getter and setter to sth like `shouldMergeUseRecordPosition` and `setMergeUseRecordPosition` so it indicates this is used for controlling the merging behavior.